hazelcast-sql/src/main/java/com/hazelcast/sql/impl/calcite/optimizer/SqlOptimizer.java

```diff
@@ -14,77 +14,50 @@
  * limitations under the License.
  */
 
-package com.hazelcast.sql.impl.calcite;
+package com.hazelcast.sql.impl.calcite.optimizer;
 
-import com.hazelcast.cluster.memberselector.MemberSelectors;
 import com.hazelcast.internal.util.collection.PartitionIdSet;
 import com.hazelcast.partition.Partition;
 import com.hazelcast.spi.impl.NodeEngine;
 import com.hazelcast.sql.impl.QueryParameterMetadata;
+import com.hazelcast.sql.impl.calcite.ExecutionContext;
 import com.hazelcast.sql.impl.calcite.opt.logical.LogicalRel;
 import com.hazelcast.sql.impl.calcite.opt.physical.PhysicalRel;
 import com.hazelcast.sql.impl.calcite.opt.physical.visitor.NodeIdVisitor;
 import com.hazelcast.sql.impl.calcite.opt.physical.visitor.PlanCreateVisitor;
 import com.hazelcast.sql.impl.calcite.opt.physical.visitor.SqlToQueryType;
 import com.hazelcast.sql.impl.calcite.parse.QueryParseResult;
-import com.hazelcast.sql.impl.optimizer.OptimizationTask;
-import com.hazelcast.sql.impl.optimizer.SqlOptimizer;
 import com.hazelcast.sql.impl.plan.Plan;
-import com.hazelcast.sql.impl.schema.TableResolver;
-import com.hazelcast.sql.impl.schema.map.PartitionedMapTableResolver;
-import com.hazelcast.sql.impl.schema.map.ReplicatedMapTableResolver;
 import com.hazelcast.sql.impl.type.QueryDataType;
 import org.apache.calcite.rel.RelNode;
 import org.apache.calcite.rel.type.RelDataType;
 
-import java.util.ArrayList;
 import java.util.Collection;
 import java.util.LinkedHashMap;
 import java.util.List;
 import java.util.Map;
 import java.util.UUID;
 
-/**
- * Calcite-based SQL optimizer.
- */
-@SuppressWarnings("checkstyle:ClassDataAbstractionCoupling")
-public class CalciteSqlOptimizer implements SqlOptimizer {
+public class SqlOptimizer {
     /** Node engine. */
     private final NodeEngine nodeEngine;
 
-    public CalciteSqlOptimizer(NodeEngine nodeEngine) {
+    public SqlOptimizer(NodeEngine nodeEngine) {
```

Comment:
I propose to avoid renames of the central classes unless there is a strong reason to. In this specific case, parsing is a part of optimization process, not vice versa.

Reply:
I can avoid renames but it is going to be awkward.
Regarding what is part of what. I have extracted parsing part from `CalciteSqlOptimizer`, there is nothing to optimize in DDL statements, thus created `SqlParser` that is able to convert a sql string into an executable statement, part of that conversion for DQLs is optimization. Not sure how else it could be done.